### PR TITLE
build CLI binary as multi stage build, instead of copying from context.

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+
+WORKDIR /app/
+
+# Install dependencies
+COPY go.mod go.sum ./
+RUN go mod download
+
+COPY . .
+
+RUN make
```
Comment (asalkeld) [inline Dockerfile]:
```suggestion
RUN make build
```
just to not rely on the default target

**Review by asalkeld — COMMENTED** [review]
small suggestion

File: pkg/provider/pulumi/generator.go [hunk 2]
```diff
@@ -127,9 +125,7 @@ func (p *pulumiDeployment) load(log output.Progress) (*auto.Stack, error) {
 	for _, plug := range p.prov.Plugins() {
 		log.Busyf("Installing Pulumi plugin %s:%s", plug.Name, plug.Version)
 
-		err = retry.Do(func() error {
-			return s.Workspace().InstallPlugin(ctx, plug.Name, plug.Version)
-		}, retry.Attempts(3), retry.Delay(time.Second))
```
Comment (asalkeld) [inline pkg/provider/pulumi/generator.go]: merge conflict?

Comment (tjholm) [reply pkg/provider/pulumi/generator.go]: looks like it, will drop this change

**Review by asalkeld — APPROVED** [review]

**Pull request merged**